app/lib/domain_events/event.rb

```diff
@@ -75,7 +75,9 @@ def self.sns_topic
     unless @sns_topic
       topic_arn = ENV.fetch('DOMAIN_EVENTS_TOPIC_ARN')
       aws_region = extract_region(topic_arn)
-      @sns_topic = Aws::SNS::Resource.new(region: aws_region).topic(topic_arn)
+      localstack_url = ENV['LOCALSTACK_URL']
+      client = Aws::SNS::Client.new(endpoint: localstack_url, region: aws_region) if localstack_url
+      @sns_topic = Aws::SNS::Resource.new(region: aws_region, client: client).topic(topic_arn)
```

Comment:
if the localstack_url is not set, then client will be nil and it works the same ways as before?

Reply:
Yes, it does.